### PR TITLE
Serialization of native trajectory iterator

### PR DESCRIPTION
**Pull request by pleroy**
Somewhat preliminary in the sense that the serialization functions can only be called with parameters that verify tight constraints.

### CODE REVIEW

File: physics/trajectory.hpp [hunk 1]
```diff
@@ -190,10 +202,25 @@ class Trajectory {
...
+    static void ReadFromMessage(
+        serialization::Trajectory::Iterator const& message,
+        not_null<Trajectory const*> const trajectory,
+        not_null<Iterator*> const iterator);
+
    private:
+    // |ancestry_| has one more element than |forks_|.  The first element in
+    // |ancestry_| is the root.  There is no element in |forks_| for the root.
+    // It is therefore empty for a root trajectory.
     typename Timeline::const_iterator current_;
```
Comment (eggrobin) [inline physics/trajectory.hpp]: comment that `current_` is never past the end

Comment (eggrobin) [inline physics/trajectory.hpp]: or maybe it can be

File: serialization/physics.proto [hunk 2]
```diff
@@ -29,14 +29,19 @@ message OblateBody {
...
-    required Point fork_time = 1;
-    repeated Trajectory trajectories = 2;
-  }
   message InstantaneousDegreesOfFreedom {
     required Point instant = 1;
     required Pair degrees_of_freedom = 2;
   }
+  message Iterator {
+    repeated int32 children_distance = 1;
+    repeated int32 timeline_distance = 2;
```
Comment (eggrobin) [inline serialization/physics.proto]: use a repeated foork

File: physics/trajectory.hpp (hunk 1 above)
Comment (pleroy) [inline physics/trajectory.hpp]: Yeah, it can be when the iterator is `at_end()`.

File: serialization/physics.proto (hunk 2 above)
Comment (pleroy) [inline serialization/physics.proto]: Done.

**Comment by pleroy** [comment]
Renamed `oork -> ork` and `Fork -> NewFork`.

**Pull request merged**